### DISCUSSION TITLE
Not able to find requested test

### DISCUSSION THREAD

**Discussion by krishnatejwani** [raysalemi/Python4RTLVerification#12] [Q&A] [answered]
Hello,

I am working on pyuvm based testbench to verify a small chip for which I am making my own bfm.

I have tried running the HelloWorldTest you gave in the book(python for RTL verification) in Figure 1. The Basic Pyuvm use model in HelloWorldTest in chapter uvm_test : Testbench: 3.0

But I am getting an error which 

"Failed to find requested test %s" % test_name
ImportError: Failed to find requested test hello_world_test
Traceback (most recent call last):

Can you please help to resolve this issue?

Thanks,
Krishna

**Comment by raysalemi** [comment]
> Hello,
> 
> 
> 
> I am working on pyuvm based testbench to verify a small chip for which I am making my own bfm.
> 
> 
> 
> I have tried running the HelloWorldTest you gave in the book(python for RTL verification) in Figure 1. The Basic Pyuvm use model in HelloWorldTest in chapter uvm_test : Testbench: 3.0
> 
> 
> 
> But I am getting an error which 
> 
> 
> 
> "Failed to find requested test %s" % test_name
> 
> ImportError: Failed to find requested test hello_world_test
> 
> Traceback (most recent call last):
> 
> 
> 
> Can you please help to resolve this issue?
> 
> 
> 
> Thanks,
> 
> Krishna

Hi Krishna,

When you say you tried running it, do you mean that you tried running the downloaded example?

Ray

**Reply by krishnatejwani** [reply]
No, I am trying to run the below code in my standalone setup.

import cocotb
import pyuvm
from   pyuvm import *

@pyuvm.test()
class hello_world_test(uvm_test) :
  async def run_phase(self) :
    self.raise_objection()
    self.logger.info("Hello, World.")
    self.drop_objection()

**Reply by raysalemi** [reply]
Could you reformat that entry using the code guards?

Three backticks above and below the code.

**Reply by raysalemi** [reply]
> No, I am trying to run the below code in my standalone setup.
> 
> 
> 
> import cocotb
> 
> import pyuvm
> 
> from   pyuvm import *
> 
> 
> 
> @pyuvm.test()
> 
> class hello_world_test(uvm_test) :
> 
>   async def run_phase(self) :
> 
>     self.raise_objection()
> 
>     self.logger.info("Hello, World.")
> 
>     self.drop_objection()
> 
> 

Please try the downloaded examples to test your environment.

**Reply by krishnatejwani** [reply]
```
import cocotb
import pyuvm
from   pyuvm import *

@pyuvm.test()
class hello_world_test(uvm_test) :
  async def run_phase(self) :
    self.raise_objection()
    self.logger.info("Hello, World.")
    self.drop_objection()
```

I am trying to run the above code.

**Comment by raysalemi** [comment]
Are you able to run the examples?

**Reply by krishnatejwani** [reply]
I tried to run the tinyalu pyuvm example using your Make file it is running all the tests specified with @pyuvm.test decorators.
However, when I am trying to run the individual test case using `make TESTCASE=AluTest` it throws an error.
Please note that I am able to run an individual test case using cocotb by invoking the same command.
![image](https://user-images.githubusercontent.com/22576610/234823064-7e247905-0b56-455d-8d8a-29af8ca848df.png)

**Comment by raysalemi** [comment]
Then you have an environment problem, not a code problem.  What message are
you getting?

**Reply by krishnatejwani** [reply]
By running `make TESTCASE=AluTest` I am getting the below error.
`"Failed to find requested test %s" % test_name
ImportError: Failed to find requested test AluTest`

**Comment by raysalemi** [comment]
Have you tried just using `make`

`TESTCASE` is not supported.

**Reply by krishnatejwani** [reply]
By Running `make AluTest` I am observing the below error.

![image](https://user-images.githubusercontent.com/22576610/234846764-e81b3e80-56e8-4398-9729-e6456183b8f8.png)

**Comment by raysalemi** [comment]
What happens if you just type `make` in the same directory as the `makefile`

**Reply by krishnatejwani** [reply]
It runs all the tests.

**Comment by raysalemi** [comment]
There you go. Then this is installed properly and is working.

What are you trying to do?

**Reply by krishnatejwani** [reply]
How can we run a single test? How can we control which test to run from command line?

**Comment by raysalemi** [comment, marked as answer]
To create Python tests that can be run using TESTCASE use the technique
described here

https://github.com/pyuvm/pyuvm/releases/tag/2.9.0

Essentially, the `pyuvm.test()` decorator is incompatible with `TESTCASE`,
but you can use the `cocotb.test()` decorator.

Ray

**Reply by krishnatejwani** [reply]
Thanks, Ray it is working now with the suggested solution.
